### PR TITLE
feat: add channel id field for Channel

### LINKED ISSUE #189: add channel id field for Channel
add a field `ID` for `Channel` which indicates the real channel id in blockchain network.

keep `metadat.name` in 
- conneciton profile's configmap name
- rbac

use `channel.spec.id` filed in 
- channel creation
- channel join
- chaincode deployment

### PR DESCRIPTION
**Pull request by 0xff-dev**
feat: add channel id field for Channel
Fix: https://github.com/bestchains/fabric-operator/issues/189

### CODE REVIEW

File: api/v1beta1/channel_webhook.go [hunk 1]
```diff
@@ -208,3 +215,23 @@ func validatePeersOwnership(ctx context.Context, c client.Client, ownerOrgs []st
...
+	selector = selector.Add(*req)
+	listOption := client.ListOptions{
+		LabelSelector: selector,
+	}
+	channelList := &ChannelList{}
+	if err := c.List(ctx, channelList, &listOption); err != nil {
+		return err
+	}
+	for _, ch := range channelList.Items {
+		klog.V(5).Infof("by selector %s, get ch %s\n", selector.String(), ch.GetName())
```
Comment (Abirdcfly) [inline api/v1beta1/channel_webhook.go:231]: At present, the V series functions do not seem to be effective. Should we temporarily remove the V function?

File: pkg/offering/base/chaincode/install.go [hunk 2]
```diff
@@ -114,7 +114,7 @@ func (c *baseChaincode) InstallChaincode(instance *current.Chaincode) (string, e
 	var finalErr error
 	for _, peer := range ch.Status.PeerConditions {
 		if peer.Type != current.PeerJoined {
-			log.Info(fmt.Sprintf("%s peer node %s has not yet joined the channel %s", method, peer.Name, instance.Spec.Channel))
+			log.Info(fmt.Sprintf("%s peer node %s has not yet joined the channel %s", method, peer.Name, ch.GetChannelID()))
```
Comment (Abirdcfly) [inline pkg/offering/base/chaincode/install.go:117]: print channel name?

File: pkg/offering/base/orderer/orderer.go [hunk 3]
```diff
@@ -514,8 +514,13 @@ func (o *Orderer) GenerateGenesisBlock(instance *current.IBPOrderer, addHostPort
...
 		}
 	}
+	ch := &current.Channel{}
+	if err := o.Client.Get(context.TODO(), types.NamespacedName{Name: instance.Spec.SystemChannelName}, ch); err != nil {
+		log.Error(err, "")
+		return nil, err
+	}
 
-	genesisBlock, err := initProfile.GenerateBlock(instance.Spec.SystemChannelName, mspConfigs)
+	genesisBlock, err := initProfile.GenerateBlock(ch.GetChannelID(), mspConfigs)
```
Comment (Abirdcfly) [inline pkg/offering/base/orderer/orderer.go:523]: no idea about this change.

File: api/v1beta1/chaincode.go [hunk 4]
```diff
@@ -34,3 +43,23 @@ func NextCond(instance *Chaincode) ChaincodeConditionType {
 	}
 	return condOrder[exp]
 }
+
+// GetChannel
+func (i *Chaincode) GetChannel(cli client.Client, cli2 controllerclient.Client) (*Channel, error) {
```
Comment (Abirdcfly) [inline api/v1beta1/chaincode.go:48]: If just want Get() function, just one "sigs.k8s.io/controller-runtime/pkg/client".Reader is enough.

File: pkg/offering/base/orderer/orderer.go [hunk 5]
```diff
@@ -516,6 +516,7 @@ func (o *Orderer) GenerateGenesisBlock(instance *current.IBPOrderer, addHostPort
 	}
 
 	genesisBlock, err := initProfile.GenerateBlock(instance.Spec.SystemChannelName, mspConfigs)
+
```
Comment (Abirdcfly) [inline pkg/offering/base/orderer/orderer.go:519]: nit:delete this line

File: pkg/offering/openshift/console/override/consolecm_test.go [hunk 6]
```diff
@@ -143,9 +143,11 @@ func CommonConsoleCMOverrides(instance *current.IBPConsole, config *v1.ConsoleSe
 		Expect(config.DBCustomNames.System).To(Equal(instance.Spec.System))
 	})
 
-	By("setting system channel", func() {
```
Comment (Abirdcfly) [inline pkg/offering/openshift/console/override/consolecm_test.go:146]: nit: restore these system channel test

File: pkg/offering/k8s/console/override/consolecm_test.go [hunk 7]
```diff
@@ -143,9 +143,11 @@ func CommonConsoleCMOverrides(instance *current.IBPConsole, config *v1.ConsoleSe
 		Expect(config.DBCustomNames.System).To(Equal(instance.Spec.System))
 	})
 
-	By("setting system channel", func() {
```
Comment (Abirdcfly) [inline pkg/offering/k8s/console/override/consolecm_test.go:146]: nit: restore these system channel test

File: pkg/offering/base/console/override/consolecm_test.go [hunk 8]
```diff
@@ -119,9 +119,9 @@ var _ = Describe("Base Console Config Map Overrides", func() {
 				Expect(config.DBCustomNames.System).To(Equal(instance.Spec.System))
 			})
 
-			By("setting system channel", func() {
```
Comment (Abirdcfly) [inline pkg/offering/base/console/override/consolecm_test.go:122]: nit: restore these system channel test

**Review by Abirdcfly — APPROVED** [review]

**Review by bjwswang — APPROVED** [review]
/lgtm

**Comment by bjwswang** [comment]
@0xff-dev pls work with @dayuy on bc-console/bc-apis changes

**Pull request merged**